shared/reducers/router.js

```diff
@@ -53,7 +54,25 @@ export function subReducer (state: RouterState = initialState, action: any): Rou
     // or a child of it
     // we get rid of everything after it
     case RouterConstants.navigateUp:
-      return state.update('uri', uri => uri.count() > 1 ? uri.pop() : uri)
+      const uri = state.get('uri')
+      if (uri.count() > 1) {
+        if (action.payload.till) {
+          let current = uri
+          while (current.count() > 0 && !_.isEqual(current.last(), action.payload.till)) { // eslint-disable-line eqeqeq
```

Comment:
You should use Immutable's `.is` https://facebook.github.io/immutable-js/docs/#/is